agent/xds/server.go

```diff
@@ -235,8 +235,11 @@ func (s *Server) process(stream ADSStream, reqCh <-chan *envoy.DiscoveryRequest)
 			return err
 		}
 
-		if rule != nil && !rule.ServiceWrite(cfgSnap.Proxy.DestinationServiceName, nil) {
-			return status.Errorf(codes.PermissionDenied, "permission denied")
+		switch cfgSnap.Kind {
+		case structs.ServiceKindConnectProxy:
+			if rule != nil && !rule.ServiceWrite(cfgSnap.Proxy.DestinationServiceName, nil) {
+				return status.Errorf(codes.PermissionDenied, "permission denied")
+			}
```

Comment:
Should this get a default block that returns permission denied on `rule != nil`?

Reply:
I updated to return an Internal error instead of perm denied for the default case.

It will have roughly the same effect of killing the ADS stream.